### PR TITLE
Use InputChanges for tasks extending SourceTask

### PR DESCRIPTION
**Pull request by wolfs**
The PR use the new incremental API (aka `InputChanges`) for the incremental tasks `AntlrTask` and `TwirlCompile`, both which are extending `SourceTask`.

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/api/tasks/SourceTask.java [hunk 1]
```diff
@@ -194,4 +203,20 @@ public SourceTask setExcludes(Iterable<String> excludes) {
...
+     *
+     * <p>
+     *     Should be used in combination with {@link org.gradle.work.InputChanges#getFileChanges(Object)} to query file changes.
+     *     For this to work, the subclass needs to override this property and add an input file annotation to it, and override {@link #getSource()} and add {@link Internal} to it.
+     * </p>
+     *
+     * @since 5.4
+     */
+    @Incubating
+    @Internal
```
Comment (lptr) [inline subprojects/core/src/main/java/org/gradle/api/tasks/SourceTask.java]: Should we mark this as `@Incremental` and `@InputFiles` and `@PathSensitive`, and mark `getSource()` as `@Internal`?

File: subprojects/core/src/test/groovy/org/gradle/api/tasks/SourceTaskTest.groovy [hunk 2]
```diff
@@ -40,5 +40,33 @@ class SourceTaskTest extends AbstractTaskTest {
...
+        File file3 = temporaryFolder.file('file3.txt').createFile()
+        file2.createNewFile()
+        file3.createNewFile()
+
+        task.source = file1
+        task.source = task.source + project.layout.files(file2) + task.source + project.layout.files(file3)
+
+        expect:
+        task.source.asList() == [file1, file2, file3]
+    }
```
Comment (lptr) [inline subprojects/core/src/test/groovy/org/gradle/api/tasks/SourceTaskTest.groovy]: Should this test test the contents of `sourceProperty` too?

File: subprojects/core/src/main/java/org/gradle/api/tasks/SourceTask.java (hunk 1 above)
Comment (wolfs) [reply subprojects/core/src/main/java/org/gradle/api/tasks/SourceTask.java]: That would possibly break the tasks which extends `SourceTask` and overwrite the path sensitivity of `getSource` with something more lenient than `@PathSensitive(RELATIVE)`.

I fail to see what the advantage of doing so would be.

**Comment by wolfs** [comment]
I don't see how we can deprecate `setSource` without having a replacement. And we can't introduce the replacement without having the new type.

I mean we could introduce a new method `replaceSource(...)` or some other `set...` method, though then we would need to override it again later on.

**Comment by wolfs** [comment]
We are not going to do the change above and try to come up with a new strategy.

**Pull request closed without merging**